include/rocksdb/db.h

```diff
@@ -353,21 +353,41 @@ class DB {
   // the i'th returned status will have Status::ok() true, and (*values)[i]
   // will store the value associated with keys[i].
   //
-  // (*values) will always be resized to be the same size as (keys).
+  // (*values) is required to be the same size as (keys).
   // Similarly, the number of returned statuses will be the number of keys.
   // Note: keys will not be "de-duplicated". Duplicate keys will return
   // duplicate values in order.
   virtual std::vector<Status> MultiGet(
       const ReadOptions& options,
       const std::vector<ColumnFamilyHandle*>& column_family,
-      const std::vector<Slice>& keys, std::vector<std::string>* values) = 0;
+      const std::vector<Slice>& keys, std::vector<std::string>* values) {
```

Comment:
Do we still care about the performance of the old interface. We introduced quite some allocations in this function. Is it OK?

Reply:
An on-stack vector does not sound much to me, compare to the cost of each DB::Get.

Reply:
on-stack vector also calls a malloc.

Reply:
@miasantreble What is your take on this?

Reply:
@maysamyabandeh on-stack vector still allocate the actual element array using allocation.